lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs

```diff
@@ -2824,22 +2820,8 @@ instance Arbitrary (PartialTx Cardano.AlonzoEra) where
         | tx' <- shrinkTx tx
         ]
       where
-        shrinkInputs (i:ins) =
-            map (:ins) (shrink i) ++ map (i:) (shrinkInputs ins)
-        shrinkInputs [] =
-            []
-
-resolvedInputsUTxO
-    :: ShelleyBasedEra era
-    -> PartialTx era
-    -> Cardano.UTxO era
-resolvedInputsUTxO era (PartialTx _ resolvedInputs _) =
-    Cardano.UTxO $ Map.fromList $ map convertUTxO resolvedInputs
-  where
-    convertUTxO (i, o, Nothing) =
-        (toCardanoTxIn i, toCardanoTxOut era o)
-    convertUTxO (_, _, Just _) =
-        error "resolvedInputsUTxO: todo: handle datum hash"
+        -- TODO: Add a shrinker
+        shrinkUTxOPreserveShape (Cardano.UTxO _) = []
```

Comment:
@Anviking, do you plan to add a shrinker in this PR?

(I guess it's difficult to claim that it preserves the shape if it replaces every UTxO set with the empty set! 😄)

Reply:
> if it replaces every UTxO set with the empty set! 😄)

Oh but that it doesn't do! The claim would be vacuously true when it doesn't actually shrink 😄 

> do you plan to add a shrinker in this PR?

🤔 I probably should have another look here, _however_ in general I am happy with taking on some short-to-medium-term technical debt in this PR.

Reply:
Some improvements: https://github.com/input-output-hk/cardano-wallet/pull/3255/commits/d524718044a91759ccaac54f9e75e9dc97398bb7